src/catching_show.c

```diff
@@ -0,0 +1,327 @@
+#define POINTS_LOST_PER_SECOND 2
+#define WEIGHT_NO_ENCOUNTER    20
+#define DISTINCT_TYPE_BONUS    50
+#define DIFFERENT_TYPE_BONUS   200
+#define MAX_TIME_SECONDS       1000
+
+typedef struct CatchingShowPokemon {
+    u16 species;
+    u8 area;
+    u8 rarity;
+    u16 catchingPoints;
+    u8 type1;
+    u8 type2;
+} CatchingShowPokemon;
+
+typedef struct CatchingShow {
+    CatchingShowPokemon pokemon[PARTY_SIZE];
+    u8 caughtMonsOrder[PARTY_SIZE];
+    s32 steps;
+    s32 currentEncounterIndex;
+    s64 startTime; // 40
+    int timePoints;
+} CatchingShow;
+
+static CatchingShow sCatchingShow;
+
+static void BufferSpeciesData(u16 species, u8 *dest);
+static void InitSpeciesData(FieldSystem *fieldSystem, CatchingShow *catchingShow);
+static int NumMonsCaptured(CatchingShow *catchingShow);
+static void ResetStepCount(CatchingShow *catchingShow);
+static BOOL IsStepCountZero(CatchingShow *catchingShow);
+static BOOL TryStartEncounter(FieldSystem *fieldSystem, CatchingShow *catchingShow, int x, int z);
+static BattleSetup *SetupEncounter(FieldSystem *fieldSystem, CatchingShow *catchingShow);
+static void UpdateBattleResultInternal(FieldSystem *fieldSystem, BattleSetup *setup, CatchingShow *catchingShow);
+static int CalcCatchingPoints(CatchingShow *catchingShow);
+static u32 CalculateTypePoints(CatchingShow *catchingShow);
+static int GetTimePoints(CatchingShow *catchingShow);
+
+void PalPark_ClearState(FieldSystem *fieldSystem) {
+    s32 i;
+    CatchingShow *local = &sCatchingShow;
+    for (i = 0; i < PARTY_SIZE; ++i) {
+        local->caughtMonsOrder[i] = 0;
+    }
+}
+
+void CatchingShow_Start(FieldSystem *fieldSystem) {
+    MI_CpuClearFast(&sCatchingShow, sizeof sCatchingShow);
+    InitSpeciesData(fieldSystem, &sCatchingShow);
+    ResetStepCount(&sCatchingShow);
+    sCatchingShow.startTime = GF_RTC_DateTimeToSec();
+}
+
+void CatchingShow_End(FieldSystem *fieldSystem) {
+    CatchingShow *catchingShow = &sCatchingShow;
+    GameStats *stats = Save_GameStats_Get(fieldSystem->saveData);
+    s64 elapsedTime = GF_RTC_TimeDelta(catchingShow->startTime, GF_RTC_DateTimeToSec());
+
+    if (elapsedTime < MAX_TIME_SECONDS) {
+        catchingShow->timePoints = ((MAX_TIME_SECONDS - elapsedTime) * POINTS_LOST_PER_SECOND);
+    } else {
+        catchingShow->timePoints = 0;
+    }
+
+    GameStats_AddScore(stats, SCORE_EVENT_17);
+}
+
+BOOL CatchingShow_CheckWildEncounter(FieldSystem *fieldSystem, int playerX, int playerY) {
+    if (IsStepCountZero(&sCatchingShow) == TRUE) {
+        return TryStartEncounter(fieldSystem, &sCatchingShow, playerX, playerY);
+    }
+    return FALSE;
+}
+
+BattleSetup *CatchingShow_GetBattleDTO(FieldSystem *fieldSystem) {
+    return SetupEncounter(fieldSystem, &sCatchingShow);
+}
+
+void CatchingShow_UpdateBattleResult(FieldSystem *fieldSystem, BattleSetup *setup) {
+    return UpdateBattleResultInternal(fieldSystem, setup, &sCatchingShow);
+}
+
+int CatchingShow_GetParkBallCount(FieldSystem *fieldSystem) {
```

Comment:
Naming:
Etc that CatchingShow_ as a prefix to a function that doesn't take the CatchingShow struct as an argument is confusion